### ISSUE TITLE
How to use predefined validation set?

### ISSUE THREAD

**Issue by vr25** [skorch-dev/skorch#624]
Hi,

I just read the issue posted [here](https://github.com/skorch-dev/skorch/issues/622#) but it still doesn't answer my question.

I want to use a predefined validation set for sklearn SVM similar to PyTorch way of training a neural network model. Basically, a Neural Net trains and validates on the train data and valid data (say, defined beforehand) to find the best model and finally, this model is used on the test data. I am trying to follow the exact approach with SVM but I am not quite clear if we can do this for non-neural methods.

Thanks!

**Comment by BenjaminBossan** [comment]
Hi,

I hope I understood your question correctly. You want to predefine a split and then use it with SVM. This has nothing to do with skorch, since SVM is from scikit-learn. Neverless, here is what you could do:

Use [`train_test_split`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) from sklearn to make a single split into train and valid, then you can train your SVM on the training data and validate on the valid date.

Alternatively, since you're talking about predefined splits, you can use sklearn's [`PredefinedSplit`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.PredefinedSplit.html) cross validator in conjunction, for instance, with [`cross_validate`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_validate.html) if you want to test more than one split.

Again, if I understand correctly, this is unrelated to skorch and you will have a better chance finding help when looking for sklearn solutions.

**Comment by BenjaminBossan** [comment]
I assume the question has been answered since there was no follow-up for a month. If that's wrong, please feel free to re-open, @vr25, and elaborate on your question.
